src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,127 @@
+package duke;
+
+import java.util.ArrayList;
+
+public class TaskList {
+
+    protected static final UI UI = new UI();
+    protected ArrayList<Task> taskList;
+
+    TaskList() {
+        this.taskList = new ArrayList<Task>();
+    }
+
+    public ArrayList<Task> getTaskList() {
+        return taskList;
+    }
+
+    /**
+     * Prints list of tasks related to description
+     *
+     * @param description
+     */
+    public void find(String description) {
+        ArrayList<Task> relatedTaskList = new ArrayList<Task>();
+
+        for (int i = 0; i < taskList.size(); ++i) {
+            Task task = taskList.get(i);
+            if (task.getDescription().toLowerCase().matches(("(.*)" + description + "(.*)").toLowerCase())) {
```

Comment:
You can consider extracting this statement into a boolean for easier readability.
- Avoid complicated expressions (consider breaking them up into smaller pieces)
I noticed several instances of this during the review
```suggestion
boolean isCondition = task.getDescription().toLowerCase().matches(("(.*)" + description + "(.*)").toLowerCase());
if (isCondition) {
```